packages/solana/lib/utils.ts

```diff
@@ -0,0 +1,37 @@
+import { Block, Transaction, TxStatus } from '@liquality/types';
+import { BlockResponse, ParsedInstruction, ParsedTransactionWithMeta, RpcResponseAndContext, SignatureStatus } from '@solana/web3.js';
+
+export function parseBlockResponse(data: BlockResponse): Block {
+    return {
+        hash: data.blockhash,
+        timestamp: data.blockTime,
+        number: data.parentSlot + 1,
+        parentHash: data.previousBlockhash,
+        _raw: data,
+    };
+}
+
+export function parseTransactionResponse(
+    transaction: ParsedTransactionWithMeta,
+    signatureStatus?: RpcResponseAndContext<SignatureStatus>
+): Transaction {
+    const {
+        transaction: {
+            message: { instructions },
+            signatures,
+        },
+    } = transaction;
+
+    const [hash] = signatures;
+    const _instructions = instructions as ParsedInstruction[];
+
+    return {
+        hash,
+        value: _instructions?.[0]?.parsed?.info?.lamports || Number(_instructions?.[0]?.parsed?.info?.amount) || 0,
```

Comment:
can you put a comment to explain what's the two different sources of the amount value?

Reply:
Added